net_util/src/tap.rs

```diff
@@ -329,61 +333,84 @@ mod tests {
         }
     }
 
-    // Describes the outcomes we are currently interested in when parsing a packet (we use
-    // an UDP packet for testing).
-    struct ParsedPkt<'a> {
-        eth: EthernetFrame<'a, &'a [u8]>,
-        ipv4: Option<IPv4Packet<'a, &'a [u8]>>,
-        udp: Option<UdpDatagram<'a, &'a [u8]>>,
+    #[derive(Debug)]
+    struct IfaceInfo {
```

Comment:
Before continuing the review, I was just wondering, what's the purpose for this structure? I see we go through quite a number of steps to obtain the IP address and MAC address for the TAP interface, but do we really need to do that? I think it's simpler if we just set a certain MAC address for the TAP interface after it's created and before it's UP, instead of getting the existing values. For the MAC address we can execute a command such as `sudo ip link set dev <interface_name> address XX:XX:XX:XX:XX:XX`. We can chat some more about this if it doesn't make that much sense here.

Reply:
so, I was thinking that this was too much too, but the whole point was to get the MAC of the `iface` that exists on the machine because that's needed for the fabricated packets now, not really the TAP device. Is there a better way to get it?

Reply:
just for documentation; Turns out that we don't really need the `eth0` (or equivalent interface)'s IP -I mentioned I needed the MAC but that wasn't correct-. as we can just bind the UdpSocket to the TAP, and send a packet to a fake destination, then read the TAP `fd` and checkout the packet